src/libraries/System.Text.Json/src/System/Text/Json/Serialization/StackFrameObjectState.cs

```diff
@@ -14,21 +14,22 @@ internal enum StackFrameObjectState : byte
 
         StartToken,
 
-        MetadataTryReadName, // Try to move the reader to the the first $id or $ref.
-        MetadataReadName, // Read the first $id or $ref.
+        ReadAheadNameOrEndObject, // Try to move the reader to the the first $id, $ref, or the EndObject token.
```

Comment:
Removed the `Metadata` prefix on these values as the names were getting too long to read/understand quickly. I think the intent is clear in the respective call sites.